src/editor/components/elements/GeoSidebar.js

```diff
@@ -5,11 +5,168 @@ import AdvancedComponents from './AdvancedComponents';
 import PropertyRow from './PropertyRow';
 import posthog from 'posthog-js';
 import useStore from '@/store';
+import { useState, useEffect } from 'react';
+import Events from '../../lib/Events';
+
+const FlatteningShapeSelector = ({
+  entity,
+  componentName,
+  shapeEntities,
+  currentValue
+}) => {
+  const handleShapeChange = (event) => {
+    const value = event.target.value;
+    // Use AFRAME inspector to properly update the entity and trigger re-renders
+    if (window.AFRAME && window.AFRAME.INSPECTOR) {
+      AFRAME.INSPECTOR.execute('entityupdate', {
+        entity: entity,
+        component: componentName,
+        property: 'flatteningShape',
+        value: value
+      });
+    }
+  };
+
+  const handleCreateShape = () => {
+    // Check if a flattening shape already exists
+    const existingShape = document.querySelector(
+      '[data-layer-name="Geo Flattening Shape"]'
+    );
+    if (existingShape) {
+      console.log(
+        'Flattening shape already exists, using existing shape:',
+        existingShape.id
+      );
+      // Select the existing shape and update component
+      if (AFRAME.INSPECTOR) {
+        AFRAME.INSPECTOR.selectEntity(existingShape);
+      }
+      if (window.AFRAME && window.AFRAME.INSPECTOR) {
+        AFRAME.INSPECTOR.execute('entityupdate', {
+          entity: entity,
+          component: componentName,
+          property: 'flatteningShape',
+          value: existingShape.id
+        });
+      }
+      return;
+    }
+
+    // Generate a unique ID
+    const shapeId = 'flattening-shape-' + Date.now();
+
+    // Create the default flattening shape using inspector command
+    const definition = {
+      id: shapeId,
+      element: 'a-box',
+      components: {
+        'data-layer-name': 'Geo Flattening Shape',
```

Comment:
That data-layer-name shouldn't be in components but at the same level as components. Does your code work? I see you're doing above
```js
const existingShape = document.querySelector(
      '[data-layer-name="Geo Flattening Shape"]'
    );
```

Reply:
uh and also class #1170